src/gennylib/test/conventions_test.cpp

```diff
@@ -1,37 +1,112 @@
 #include "test.h"
 
 #include <chrono>
+#include <cmath>
 
 #include <yaml-cpp/yaml.h>
 
 #include <gennylib/conventions.hpp>
 
+namespace genny {
+namespace {
 using namespace std::chrono;
 
-TEST_CASE("Millisecond conversions") {
-    SECTION("Can convert to milliseconds") {
-        REQUIRE(YAML::Load("D: 300")["D"].as<milliseconds>().count() == 300);
-        REQUIRE(YAML::Load("-1").as<milliseconds>().count() == -1);
-        REQUIRE(YAML::Load("0").as<milliseconds>().count() == 0);
+TEST_CASE("genny::TimeSpec conversions") {
+    SECTION("Can convert to genny::TimeSpec") {
+        REQUIRE(YAML::Load("D: 3 seconds")["D"].as<TimeSpec>().count() == 3 * std::pow(10, 9));
+        REQUIRE(YAML::Load("0 second").as<TimeSpec>().count() == 0);
+        REQUIRE(YAML::Load("20 millisecond").as<TimeSpec>().count() == 20 * std::pow(10, 6));
+        REQUIRE(YAML::Load("33 microsecond").as<TimeSpec>().count() == 33 * std::pow(10, 3));
+        REQUIRE(YAML::Load("2e3 microseconds").as<TimeSpec>().count() == 2000 * std::pow(10, 3));
+        REQUIRE(YAML::Load("10.3e2 nanoseconds").as<TimeSpec>().count() == 1030);
+        REQUIRE(YAML::Load("3 hour").as<TimeSpec>().count() == 3 * 3600 * std::pow(10, 9));
+        REQUIRE(YAML::Load("2 minutes").as<TimeSpec>().count() == 2 * 60 * std::pow(10, 9));
```

Comment:
Just as a heads up, [pow](https://en.cppreference.com/w/cpp/numeric/math/pow) is inately floating point.